shared/src/commonMain/kotlin/com/mbta/tid/mbta_app/model/PatternSorting.kt

```diff
@@ -0,0 +1,83 @@
+package com.mbta.tid.mbta_app.model
+
+import io.github.dellisd.spatialk.geojson.Position
+
+object PatternSorting {
+    private fun patternServiceBucket(realtimePatterns: RealtimePatterns) =
+        when {
+            // service or alert today
+            realtimePatterns.hasSchedulesToday ||
+                realtimePatterns.hasMajorAlerts ||
+                realtimePatterns.upcomingTrips.orEmpty().isNotEmpty() -> 1
+            // no service today
+            else -> 2
+        }
+
+    private fun pinnedRouteBucket(route: Route, pinnedRoutes: Set<String>) =
+        when {
+            pinnedRoutes.contains(route.id) || pinnedRoutes.contains(route.lineId) -> 1
+            else -> 2
+        }
+
+    private fun subwayBucket(route: Route) =
+        when {
+            route.type.isSubway() -> 1
+            else -> 2
+        }
+
+    fun compareStaticPatterns(): Comparator<NearbyStaticData.StaticPatterns> =
+        compareBy(
+            { it.patterns.first().directionId },
+            {
+                when (it) {
+                    is NearbyStaticData.StaticPatterns.ByDirection -> -1
+                    is NearbyStaticData.StaticPatterns.ByHeadsign -> 1
+                }
+            },
+            { it.patterns.first() }
+        )
+
+    fun comparePatterns(): Comparator<RealtimePatterns> =
```

Comment:
nitpick (non-blocking): consider calling this `compareRealtimePatterns` for consistency with `compare static patterns`

Reply:
Renamed.